docs/pages/architecture/agents.mdx

```diff
@@ -0,0 +1,255 @@
+| SSH Service | SSH traffic to the target SSH Service instance or remote OpenSSH servers.|
+| Teleport Kubernetes Service  | HTTPS traffic to Kubernetes API servers.|
+| Teleport Database Service    | Database-native wire protocols such as Redis, MySQL, plus HTTP-based database protocols such as DynamoDB.|
+| Teleport Application Service | HTTPS and TCP forwarding for internal web applications, with optional JSON web tokens. Signed HTTPS messages for cloud provider APIs.|
+| Teleport Desktop Service     | Remote Desktop Protocol traffic to remote servers. Converts incoming RDP traffic to Teleport Desktop Protocol traffic before returning itto the Teleport Web UI via the browser.|
+
+## Components
+
+![Components of a Teleport agent deployment](../../img/architecture/agent-architecture.png)
+
+A Teleport cluster where an administrator has enrolled resources involves the
+following components:
+
+- **[Teleport Proxy Service](../architecture/proxy.mdx):** A stateless service that
+  performs the function of an authentication gateway, serves the Web UI, and
+  accepts client connections. On cloud-hosted Teleport Enterprise accounts, this
+  service is accessible at your Teleport account URL, e.g.,
+  `example.teleport.sh`.
+- **[Teleport Auth Service](../architecture/authentication.mdx):** Serves as
+  cluster's certificate authority, handles user authentication/authorization and
+  issues short-lived client certificates.
+- **Teleport agents:** Agents perform authentication against infrastructure
+  resources, proxy user traffic to those resources, and perform protocol
+  parsing.
+- **Teleport client tools (`tsh`, Teleport Connect, and the Teleport Web UI):**
+  Connect to resources in your infrastructure through Teleport agents and the
+  Teleport Proxy Service.
+- **Local proxies:** In some cases, `tsh` spins up local proxy servers that
+  authenticate to Teleport and forward traffic from client tools (e.g., database
+  clients and AWS SDK applications).
+- **Infrastructure resources:** Teleport can protect access to self-hosted
+  infrastructure as well as infrastructure managed by a cloud provider.
+
+## Retrieving credentials from the Teleport Auth Service
+
+The Teleport Auth Service issues certificates that enable users to communicate
+with Teleport agents and for Teleport agents to communicate with resources in
+your infrastructure.
+
+### Credentials for Teleport agents
+
+The Teleport Auth Service runs a certificate authority that issues a host
+certificate to an agent when it joins the cluster for the first time. Read [Join
+Services to your Teleport Cluster](../agents/join-services-to-your-cluster.mdx)
+for the available methods you can use to join an agent to your Teleport cluster.
+
+All agents in a Teleport cluster keep the Auth Service updated on their status
+with periodic ping messages. They report their IP addresses and the values of
+their assigned labels. Clients can access the list of all agent instances in
```

Comment:
@greedy52 My understanding is that all services connect and stay connected in a similar way. @rosstimothy knows more about the details of this, if I'm not mistaken.

I'm also unclear if this is done on a per agent basis, or per service basis.

Reply:
I've removed this sentence so we can get this merged more quickly. It looks like the heartbeat system has some intricacies that might make more sense to skip for now instead of delve into.